### PR TITLE
[AV-127117] Make CopyToRegions a Set

### PR DESCRIPTION
**Pull request by SophieWegmann**
## Jira

* AV-127117

## Description

Change `CopyToRegions` from `[]string` to `types.Set`, so there won't be an error if `CopyToRegions` is unknown. Make `CopyToRegions` null in the `createSnapshotBackupScheduleRequest`, if it is unknown. 

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue).
- [ ] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] This change updates the ci/cd workflow.
- [ ] Documentation fix/enhancement.

### How was this change tested and do you have evidence? _**(REQUIRED: Select at least 1)**_

- [x] Manually tested
- [ ] Unit tested
- [x] Acceptance tested
- [ ] Unable to test / will not test (Please provide comments in section below)

### Testing

<details open>
  <summary>Testing</summary>

**Snapshot Backup Schedule Resource for a GCP Cluster**

*Create* 

Running `terraform plan` with 

```
cloud_snapshot_backup_schedule = {
    interval = 4
    retention = 168
}
```
creates the plan 

```
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # couchbase-capella_cloud_snapshot_backup_schedule.new_cloud_snapshot_backup_schedule will be created
  + resource "couchbase-capella_cloud_snapshot_backup_schedule" "new_cloud_snapshot_backup_schedule" {
      + cluster_id      = "ffffffff-aaaa-1414-eeee-000000000000"
      + copy_to_regions = (known after apply)
      + interval        = 4
      + organization_id = "ffffffff-aaaa-1414-eeee-000000000000"
      + project_id      = "ffffffff-aaaa-1414-eeee-000000000000"
      + retention       = 168
      + start_time      = "2026-04-09T16:00:00+01:00"
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + new_cloud_snapshot_backup_schedule = {
      + cluster_id      = "ffffffff-aaaa-1414-eeee-000000000000"
      + copy_to_regions = (known after apply)
      + interval        = 4
      + organization_id = "ffffffff-aaaa-1414-eeee-000000000000"
      + project_id      = "ffffffff-aaaa-1414-eeee-000000000000"
      + retention       = 168
      + start_time      = "2026-04-09T16:00:00+01:00"
    }
```

Running `terraform apply` successfully creates the Snapshot Backup Schedule resource.

```
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # couchbase-capella_cloud_snapshot_backup_schedule.new_cloud_snapshot_backup_schedule will be created
  + resource "couchbase-capella_cloud_snapshot_backup_schedule" "new_cloud_snapshot_backup_schedule" {
      + cluster_id      = "ffffffff-aaaa-1414-eeee-000000000000"
      + copy_to_regions = (known after apply)
      + interval        = 4
      + organization_id = "ffffffff-aaaa-1414-eeee-000000000000"
      + project_id      = "ffffffff-aaaa-1414-eeee-000000000000"
      + retention       = 168
      + start_time      = "2026-04-09T16:00:00+01:00"
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + new_cloud_snapshot_backup_schedule = {
      + cluster_id      = "ffffffff-aaaa-1414-eeee-000000000000"
      + copy_to_regions = (known after apply)
      + interval        = 4
      + organization_id = "ffffffff-aaaa-1414-eeee-000000000000"
      + project_id      = "ffffffff-aaaa-1414-eeee-000000000000"
      + retention       = 168
      + start_time      = "2026-04-09T16:00:00+01:00"
    }

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

couchbase-capella_cloud_snapshot_backup_schedule.new_cloud_snapshot_backup_schedule: Creating...
couchbase-capella_cloud_snapshot_backup_schedule.new_cloud_snapshot_backup_schedule: Creation complete after 0s

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

new_cloud_snapshot_backup_schedule = {
  "cluster_id" = "ffffffff-aaaa-1414-eeee-000000000000"
  "copy_to_regions" = toset(null) /* of string */
  "interval" = 4
  "organization_id" = "ffffffff-aaaa-1414-eeee-000000000000"
  "project_id" = "ffffffff-aaaa-1414-eeee-000000000000"
  "retention" = 168
  "start_time" = "2026-04-09T16:00:00+01:00"
}
```

Screenshot 2026-04-09 at 16 28 35

*Update*

Running `terraform apply` with 

```
cloud_snapshot_backup_schedule = {
    interval = 4
    retention = 168
   copy_to_regions = ["test"]
}
```

returns an error message explaining that cross-region copies are not supported for GCP clusters.

```
couchbase-capella_cloud_snapshot_backup_schedule.new_cloud_snapshot_backup_schedule: Refreshing state...

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # couchbase-capella_cloud_snapshot_backup_schedule.new_cloud_snapshot_backup_schedule will be updated in-place
  ~ resource "couchbase-capella_cloud_snapshot_backup_schedule" "new_cloud_snapshot_backup_schedule" {
      + copy_to_regions = [
          + "test",
        ]
        # (6 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.

Changes to Outputs:
  ~ new_cloud_snapshot_backup_schedule = {
      ~ copy_to_regions = null -> [
          + "test",
        ]
        # (6 unchanged attributes hidden)
    }

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

couchbase-capella_cloud_snapshot_backup_schedule.new_cloud_snapshot_backup_schedule: Modifying...
╷
│ Error: Error Upserting Snapshot Backup Schedule in Capella
│ 
│   with couchbase-capella_cloud_snapshot_backup_schedule.new_cloud_snapshot_backup_schedule,
│   on create_snapshot_backup_schedule.tf line 5, in resource "couchbase-capella_cloud_snapshot_backup_schedule" "new_cloud_snapshot_backup_schedule":
│    5: resource "couchbase-capella_cloud_snapshot_backup_schedule" "new_cloud_snapshot_backup_schedule" {
│ 
│ Could not upsert Capella Snapshot Backup Schedule for cluster with ID ffffffff-aaaa-1414-eeee-000000000000: {"code":501,"hint":"The server does
│ not support the functionality required to fulfill the request. Please contact support for more
│ information.","httpStatusCode":501,"message":"Cross-region cloud snapshot backup and restore operations are not supported for GCP clusters at
│ this time. Please contact support for more information."}
╵
```

Running `terraform plan` with

```
cloud_snapshot_backup_schedule = {
    interval = 12
    retention = 168
}
```

creates the following plan

```
couchbase-capella_cloud_snapshot_backup_schedule.new_cloud_snapshot_backup_schedule: Refreshing state...

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # couchbase-capella_cloud_snapshot_backup_schedule.new_cloud_snapshot_backup_schedule will be updated in-place
  ~ resource "couchbase-capella_cloud_snapshot_backup_schedule" "new_cloud_snapshot_backup_schedule" {
      + copy_to_regions = (known after apply)
      ~ interval        = 4 -> 12
        # (5 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.

Changes to Outputs:
  ~ new_cloud_snapshot_backup_schedule = {
      + copy_to_regions = (known after apply)
      ~ interval        = 4 -> 12
        # (5 unchanged attributes hidden)
    }
```

Running `terraform apply` successfully updates the Snapshot Backup Schedule resource.

```
couchbase-capella_cloud_snapshot_backup_schedule.new_cloud_snapshot_backup_schedule: Refreshing state...

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # couchbase-capella_cloud_snapshot_backup_schedule.new_cloud_snapshot_backup_schedule will be updated in-place
  ~ resource "couchbase-capella_cloud_snapshot_backup_schedule" "new_cloud_snapshot_backup_schedule" {
      + copy_to_regions = (known after apply)
      ~ interval        = 4 -> 12
        # (5 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.

Changes to Outputs:
  ~ new_cloud_snapshot_backup_schedule = {
      + copy_to_regions = (known after apply)
      ~ interval        = 4 -> 12
        # (5 unchanged attributes hidden)
    }

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

couchbase-capella_cloud_snapshot_backup_schedule.new_cloud_snapshot_backup_schedule: Modifying...
couchbase-capella_cloud_snapshot_backup_schedule.new_cloud_snapshot_backup_schedule: Modifications complete after 1s

Apply complete! Resources: 0 added, 1 changed, 0 destroyed.

Outputs:

new_cloud_snapshot_backup_schedule = {
  "cluster_id" = "ffffffff-aaaa-1414-eeee-000000000000"
  "copy_to_regions" = toset(null) /* of string */
  "interval" = 12
  "organization_id" = "ffffffff-aaaa-1414-eeee-000000000000"
  "project_id" = "ffffffff-aaaa-1414-eeee-000000000000"
  "retention" = 168
  "start_time" = "2026-04-09T16:00:00+01:00"
}
```
Screenshot 2026-04-09 at 16 43 31

**Snapshot Backup Schedule Data Source for a GCP Cluster**

Running `terraform plan` creates the following plan

```
data.couchbase-capella_cloud_snapshot_backup_schedule.existing_cloud_snapshot_backup_schedule: Reading...
data.couchbase-capella_cloud_snapshot_backup_schedule.existing_cloud_snapshot_backup_schedule: Read complete after 0s

Changes to Outputs:
  + existing_cloud_snapshot_backup_schedule = {
      + cluster_id      = "ffffffff-aaaa-1414-eeee-000000000000"
      + copy_to_regions = null
      + interval        = 12
      + organization_id = "ffffffff-aaaa-1414-eeee-000000000000"
      + project_id      = "ffffffff-aaaa-1414-eeee-000000000000"
      + retention       = 168
      + start_time      = "2026-04-09T16:00:00+01:00"
    }

You can apply this plan to save these new output values to the Terraform state, without changing any real infrastructure.
```

Running `terraform apply` successfully applies the plan.

```
data.couchbase-capella_cloud_snapshot_backup_schedule.existing_cloud_snapshot_backup_schedule: Reading...
data.couchbase-capella_cloud_snapshot_backup_schedule.existing_cloud_snapshot_backup_schedule: Read complete after 0s

Changes to Outputs:
  + existing_cloud_snapshot_backup_schedule = {
      + cluster_id      = "ffffffff-aaaa-1414-eeee-000000000000"
      + copy_to_regions = null
      + interval        = 12
      + organization_id = "ffffffff-aaaa-1414-eeee-000000000000"
      + project_id      = "ffffffff-aaaa-1414-eeee-000000000000"
      + retention       = 168
      + start_time      = "2026-04-09T16:00:00+01:00"
    }

You can apply this plan to save these new output values to the Terraform state, without changing any real infrastructure.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes


Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

existing_cloud_snapshot_backup_schedule = {
  "cluster_id" = "ffffffff-aaaa-1414-eeee-000000000000"
  "copy_to_regions" = toset(null) /* of string */
  "interval" = 12
  "organization_id" = "ffffffff-aaaa-1414-eeee-000000000000"
  "project_id" = "ffffffff-aaaa-1414-eeee-000000000000"
  "retention" = 168
  "start_time" = "2026-04-09T16:00:00+01:00"
}
```

**Snapshot Backup Schedule Resource for an AWS Cluster**

*Create*

Running `terraform plan` with 

```
cloud_snapshot_backup_schedule = {
    interval = 4
    retention = 168
}
```

creates the following plan

```
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the
following symbols:
  + create

Terraform will perform the following actions:

  # couchbase-capella_cloud_snapshot_backup_schedule.new_cloud_snapshot_backup_schedule will be created
  + resource "couchbase-capella_cloud_snapshot_backup_schedule" "new_cloud_snapshot_backup_schedule" {
      + cluster_id      = "ffffffff-aaaa-1414-eeee-000000000000"
      + copy_to_regions = (known after apply)
      + interval        = 4
      + organization_id = "ffffffff-aaaa-1414-eeee-000000000000"
      + project_id      = "ffffffff-aaaa-1414-eeee-000000000000"
      + retention       = 168
      + start_time      = "2026-04-09T17:00:00+01:00"
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + new_cloud_snapshot_backup_schedule = {
      + cluster_id      = "ffffffff-aaaa-1414-eeee-000000000000"
      + copy_to_regions = (known after apply)
      + interval        = 4
      + organization_id = "ffffffff-aaaa-1414-eeee-000000000000"
      + project_id      = "ffffffff-aaaa-1414-eeee-000000000000"
      + retention       = 168
      + start_time      = "2026-04-09T17:00:00+01:00"
    }
```

Running `terraform apply`  successfully creates the Snapshot Backup Schedule resource.

```
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the
following symbols:
  + create

Terraform will perform the following actions:

  # couchbase-capella_cloud_snapshot_backup_schedule.new_cloud_snapshot_backup_schedule will be created
  + resource "couchbase-capella_cloud_snapshot_backup_schedule" "new_cloud_snapshot_backup_schedule" {
      + cluster_id      = "ffffffff-aaaa-1414-eeee-000000000000"
      + copy_to_regions = (known after apply)
      + interval        = 4
      + organization_id = "ffffffff-aaaa-1414-eeee-000000000000"
      + project_id      = "ffffffff-aaaa-1414-eeee-000000000000"
      + retention       = 168
      + start_time      = "2026-04-09T17:00:00+01:00"
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + new_cloud_snapshot_backup_schedule = {
      + cluster_id      = "ffffffff-aaaa-1414-eeee-000000000000"
      + copy_to_regions = (known after apply)
      + interval        = 4
      + organization_id = "ffffffff-aaaa-1414-eeee-000000000000"
      + project_id      = "ffffffff-aaaa-1414-eeee-000000000000"
      + retention       = 168
      + start_time      = "2026-04-09T17:00:00+01:00"
    }

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

couchbase-capella_cloud_snapshot_backup_schedule.new_cloud_snapshot_backup_schedule: Creating...
couchbase-capella_cloud_snapshot_backup_schedule.new_cloud_snapshot_backup_schedule: Creation complete after 1s

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

new_cloud_snapshot_backup_schedule = {
  "cluster_id" = "ffffffff-aaaa-1414-eeee-000000000000"
  "copy_to_regions" = toset(null) /* of string */
  "interval" = 4
  "organization_id" = "ffffffff-aaaa-1414-eeee-000000000000"
  "project_id" = "ffffffff-aaaa-1414-eeee-000000000000"
  "retention" = 168
  "start_time" = "2026-04-09T17:00:00+01:00"
}
```

Screenshot 2026-04-09 at 17 27 00

*Update*

Running `terraform plan` with 

```
cloud_snapshot_backup_schedule = {
    interval = 12
    retention = 168
    copy_to_regions = ["eu-west-1"]
}
```

creates the following plan 

```
couchbase-capella_cloud_snapshot_backup_schedule.new_cloud_snapshot_backup_schedule: Refreshing state...

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the
following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # couchbase-capella_cloud_snapshot_backup_schedule.new_cloud_snapshot_backup_schedule will be updated in-place
  ~ resource "couchbase-capella_cloud_snapshot_backup_schedule" "new_cloud_snapshot_backup_schedule" {
      + copy_to_regions = [
          + "eu-west-1",
        ]
      ~ interval        = 4 -> 12
        # (5 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.

Changes to Outputs:
  ~ new_cloud_snapshot_backup_schedule = {
      ~ copy_to_regions = null -> [
          + "eu-west-1",
        ]
      ~ interval        = 4 -> 12
        # (5 unchanged attributes hidden)
    }
```

Running `terraform apply` successfully updates the Snapshot Backup Schedule.

```
couchbase-capella_cloud_snapshot_backup_schedule.new_cloud_snapshot_backup_schedule: Refreshing state...

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the
following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # couchbase-capella_cloud_snapshot_backup_schedule.new_cloud_snapshot_backup_schedule will be updated in-place
  ~ resource "couchbase-capella_cloud_snapshot_backup_schedule" "new_cloud_snapshot_backup_schedule" {
      + copy_to_regions = [
          + "eu-west-1",
        ]
      ~ interval        = 4 -> 12
        # (5 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.

Changes to Outputs:
  ~ new_cloud_snapshot_backup_schedule = {
      ~ copy_to_regions = null -> [
          + "eu-west-1",
        ]
      ~ interval        = 4 -> 12
        # (5 unchanged attributes hidden)
    }

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

couchbase-capella_cloud_snapshot_backup_schedule.new_cloud_snapshot_backup_schedule: Modifying...
couchbase-capella_cloud_snapshot_backup_schedule.new_cloud_snapshot_backup_schedule: Modifications complete after 0s

Apply complete! Resources: 0 added, 1 changed, 0 destroyed.

Outputs:

new_cloud_snapshot_backup_schedule = {
  "cluster_id" = "ffffffff-aaaa-1414-eeee-000000000000"
  "copy_to_regions" = toset([
    "eu-west-1",
  ])
  "interval" = 12
  "organization_id" = "ffffffff-aaaa-1414-eeee-000000000000"
  "project_id" = "ffffffff-aaaa-1414-eeee-000000000000"
  "retention" = 168
  "start_time" = "2026-04-09T17:00:00+01:00"
}
```

Screenshot 2026-04-09 at 17 31 53

**Snapshot Backup Schedule Data Source for an AWS Cluster**

Running `terraform plan` creates the following plan

```
data.couchbase-capella_cloud_snapshot_backup_schedule.existing_cloud_snapshot_backup_schedule: Reading...
data.couchbase-capella_cloud_snapshot_backup_schedule.existing_cloud_snapshot_backup_schedule: Read complete after 1s

Changes to Outputs:
  + existing_cloud_snapshot_backup_schedule = {
      + cluster_id      = "71f78545-3d7b-46f1-8a3a-9e240211b574"
      + copy_to_regions = [
          + "eu-west-1",
        ]
      + interval        = 12
      + organization_id = "adb4fb4c-1d98-4287-ac33-230742d2cc76"
      + project_id      = "68de26a7-e86b-47b5-a1bf-e66daab92d65"
      + retention       = 168
      + start_time      = "2026-04-09T17:00:00+01:00"
    }

You can apply this plan to save these new output values to the Terraform state, without changing any real infrastructure.
```

Running `terraform apply` successfully applies the plan

```
data.couchbase-capella_cloud_snapshot_backup_schedule.existing_cloud_snapshot_backup_schedule: Reading...
data.couchbase-capella_cloud_snapshot_backup_schedule.existing_cloud_snapshot_backup_schedule: Read complete after 0s

Changes to Outputs:
  + existing_cloud_snapshot_backup_schedule = {
      + cluster_id      = "71f78545-3d7b-46f1-8a3a-9e240211b574"
      + copy_to_regions = [
          + "eu-west-1",
        ]
      + interval        = 12
      + organization_id = "adb4fb4c-1d98-4287-ac33-230742d2cc76"
      + project_id      = "68de26a7-e86b-47b5-a1bf-e66daab92d65"
      + retention       = 168
      + start_time      = "2026-04-09T17:00:00+01:00"
    }

You can apply this plan to save these new output values to the Terraform state, without changing any real infrastructure.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes


Apply complete! Resources: 0 added, 0 changed, 0 destroyed.

Outputs:

existing_cloud_snapshot_backup_schedule = {
  "cluster_id" = "71f78545-3d7b-46f1-8a3a-9e240211b574"
  "copy_to_regions" = toset([
    "eu-west-1",
  ])
  "interval" = 12
  "organization_id" = "adb4fb4c-1d98-4287-ac33-230742d2cc76"
  "project_id" = "68de26a7-e86b-47b5-a1bf-e66daab92d65"
  "retention" = 168
  "start_time" = "2026-04-09T17:00:00+01:00"
}
```

</details>

## Further comments

### CODE REVIEW

File: internal/resources/snapshot_backup_schedule.go [hunk 1]
```diff
@@ -70,7 +70,16 @@ func (s *SnapshotBackupSchedule) Create(ctx context.Context, req resource.Create
 		clusterId      = plan.ClusterID.ValueString()
 	)
 
-	err := s.upsertSnapshotBackupSchedule(ctx, organizationId, projectId, clusterId, plan)
+	var copyToRegions []string
+	if !plan.CopyToRegions.IsUnknown() {
```
[the remark this answers is not in the document]
Comment (SophieWegmann) [reply internal/resources/snapshot_backup_schedule.go]: Explicitly setting `copy_to_regions` to null eg

```
cloud_snapshot_backup_schedule = {
    interval = 12
    retention = 168
    copy_to_regions = null
}
```

and then applying the plan does not cause an error.

```
Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the
following symbols:
  + create

Terraform will perform the following actions:

  # couchbase-capella_cloud_snapshot_backup_schedule.new_cloud_snapshot_backup_schedule will be created
  + resource "couchbase-capella_cloud_snapshot_backup_schedule" "new_cloud_snapshot_backup_schedule" {
      + cluster_id      = "ffffffff-aaaa-1414-eeee-000000000000"
      + copy_to_regions = (known after apply)
      + interval        = 12
      + organization_id = "ffffffff-aaaa-1414-eeee-000000000000"
      + project_id      = "ffffffff-aaaa-1414-eeee-000000000000"
      + retention       = 168
      + start_time      = "2026-04-10T12:00:00+01:00"
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + new_cloud_snapshot_backup_schedule = {
      + cluster_id      = "ffffffff-aaaa-1414-eeee-000000000000"
      + copy_to_regions = (known after apply)
      + interval        = 12
      + organization_id = "ffffffff-aaaa-1414-eeee-000000000000"
      + project_id      = "ffffffff-aaaa-1414-eeee-000000000000"
      + retention       = 168
      + start_time      = "2026-04-10T12:00:00+01:00"
    }

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

couchbase-capella_cloud_snapshot_backup_schedule.new_cloud_snapshot_backup_schedule: Creating...
couchbase-capella_cloud_snapshot_backup_schedule.new_cloud_snapshot_backup_schedule: Creation complete after 0s

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

new_cloud_snapshot_backup_schedule = {
  "cluster_id" = "ffffffff-aaaa-1414-eeee-000000000000"
  "copy_to_regions" = toset(null) /* of string */
  "interval" = 12
  "organization_id" = "ffffffff-aaaa-1414-eeee-000000000000"
  "project_id" = "ffffffff-aaaa-1414-eeee-000000000000"
  "retention" = 168
  "start_time" = "2026-04-10T12:00:00+01:00"
}
```

File: internal/schema/snapshot_backup_schedule.go [hunk 2]
```diff
@@ -33,15 +33,15 @@ func (s SnapshotBackupSchedule) AttributeTypes() map[string]attr.Type {
 	}
 }
 
-func NewSnapshotBackupSchedule(snapshotBackupSchedule snapshot_backup_schedule.SnapshotBackupSchedule, organizationID, projectID, clusterID string) SnapshotBackupSchedule {
+func NewSnapshotBackupSchedule(snapshotBackupSchedule snapshot_backup_schedule.SnapshotBackupSchedule, organizationID, projectID, clusterID string, copyToRegions basetypes.SetValue) SnapshotBackupSchedule {
 	return SnapshotBackupSchedule{
```
Comment (matty271828) [inline internal/schema/snapshot_backup_schedule.go]: Looks like we can pull 

```
	copyToRegions, diags := types.SetValueFrom(ctx, types.StringType, scheduleResp.CopyToRegions)
	if diags.HasError() {
		return nil, fmt.Errorf("copyToRegions set error")
	}
```

into this method and remove some duplication between `morphToTerraformCloudSnapshotBackupSchedule` and the datasource read method. Can we modify this to return an error and do so?

File: internal/resources/snapshot_backup_schedule.go [hunk 3]
```diff
@@ -355,6 +398,17 @@ func (s *SnapshotBackupSchedule) getStartTime(ctx context.Context, currentStartT
 	return newStartTime.Format(time.RFC3339), nil
 }
 
+// morphToTerraformCloudSnapshotBackupSchedule creates a snapshot backup schedule from a snapshot backup schedule response.
```
Comment (matty271828) [inline internal/resources/snapshot_backup_schedule.go]: If we do https://github.com/couchbasecloud/terraform-provider-couchbase-capella/pull/548/changes#r3074536285 we should be able to remove this method?

File: internal/resources/snapshot_backup_schedule.go [hunk 4]
```diff
@@ -171,14 +196,24 @@ func (s *SnapshotBackupSchedule) Update(ctx context.Context, req resource.Update
 		clusterId      = plan.ClusterID.ValueString()
 	)
 
-	err := s.upsertSnapshotBackupSchedule(ctx, organizationId, projectId, clusterId, plan)
+	var copyToRegions []string
```
Comment (matty271828) [inline internal/resources/snapshot_backup_schedule.go]: could pull this code block out into a helper function

**Review by matty271828 — COMMENTED** [review]
Looks good just some stylistic nits. 

Is there an existing acceptance test for this feature that we could add a test case for both the populated and the null case behaviour?

**Review by matty271828 — APPROVED** [review]
Nice work!
